architecture/azure/messaging.md

```diff
@@ -0,0 +1,97 @@
+
+### Azure Service Bus
+
+[Azure Service Bus](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview) supports queuing, publish/subscribe, and more advanced integration patterns. It is designed to integrate applications or application components that may span multiple communication protocols, data contracts, trust domains, or network environments.
+
+Azure Service Bus has two pricing tiers: [Standard and Premium](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-premium-messaging). The Premium tier provides hardware level isolation, which means each customer's workload runs on dedicated hardware. Premium tier uses _messaging units_ for pricing. The Standard tier offers pay-as-you-go pricing with less predictable latency and throughput. Microsoft recommends running production systems on the Premium tier.
+
+:heavy_plus_sign: Pros:
+
+- Up to 100 MB message size on the Premium tier
+- Supports transactions across multiple queues
+- Scripted infrastructure deployment using the [NServiceBus ASB CLI](/transports/azure-service-bus/operational-scripting.md)
+- Supports the AMQP 1.0 protocol
+- Supports cross-entity transactions
+
+:heavy_minus_sign: Cons:
+
+- Some features are only available on the Premium tier
+- The maximum message size is 256 KB on the Standard tier
+- The maximum queue size is 80 GB on the Standard tier
+- There is no on-premises equivalent.
+
+[**Try the Azure Service Bus Tutorial →**](https://learn.microsoft.com/en-us/azure/service-bus-messaging/build-message-driven-apps-nservicebus?tabs=Sender)
+
+#### When to use Azure Service Bus
+
+Azure Service Bus may be considered the default messaging choice for Azure. Alternatives should be considered only when the message size limitations are not sufficient or if a more portable queueing technology is required.
+
+### Azure Queue Storage
+
+[Azure Queue Storage](https://learn.microsoft.com/en-us/azure/storage/queues/storage-queues-introduction) offer a simple, low-cost, and easy-to-use messaging service based on the Azure Storage infrastructure.
+
+:heavy_plus_sign: Pros:
+
+- Queues can contain up to 200 TB of messages
+- Cheaper than Azure Service Bus
+- Part of Azure Storage
+- The Azure Storage Emulator may be on-premises for development, testing, and CI
+
+:heavy_minus_sign: Cons:
+
+- The maximum message size is 64 KB
+- Higher latency due to HTTP communication
+- No cross-entity transactions
+- The maximum number of requests (e.g. message delivery) per storage account is 20,000 per second (2,000 per second for single queues)
+
+[**Try the Azure Storage Queues sample →**](/samples/azure/storage-queues/)
+
+#### When to use Azure Storage Queues
+
+Consider Azure Storage Queues when a managed message queuing technology is required but Azure Service Bus doesn't meet requirements or is too expensive.
```

Comment:
```suggestion
Azure Storage Queues is a legacy Azure service that is not recommended for new systems.
```

$10/m for the ASB standard tier for a commercial system running on NSB is cheaper than the licensing cost of NSB for a solution that includes 4+ endpoints, which is any NSB-based system in production.

Reply:
I've updated the suggestion. Have a look